src/function/maybe_changed_after.rs

```diff
@@ -118,14 +171,23 @@ where
         zalsa: &Zalsa,
         database_key_index: DatabaseKeyIndex,
         memo: &Memo<C::Output<'_>>,
+        allow_provisional: bool,
```

Comment:
I'd find it helpful to have some documentation around `allow_provisional`. When are provisional values allowed?